src/Quotations.ts

```diff
@@ -105,13 +105,48 @@ export function extractFromHtml(messageBody: string): ExtractFromHtmlResult {
   if (numberOfCheckpoints >= NodeLimit)
     return { body: messageBody, didFindQuote: false, isTooLong: true };
 
-  const quotationCheckpoints = new Array<boolean>(numberOfCheckpoints);
-  const messagePlainText = preprocess(elementToText(xmlDocument), "\n", ContentTypeTextPlain);
+  let extractQuoteHtml = extractQuoteHtmlViaMarkers(numberOfCheckpoints, xmlDocument, false);
+  if (extractQuoteHtml.error)
+    return { body: messageBody, didFindQuote: false, isTooLong: true };
+
+  // Make sure we did not miss a quote due to some parsing error
+  if (!extractQuoteHtml.quoteWasFound)
+    extractQuoteHtml = extractQuoteHtmlViaMarkers(numberOfCheckpoints, xmlDocument, true);
+
+  if(extractQuoteHtml.quoteWasFound) {
+    // Remove the tags that we marked as quotation from the HTML.
+    deleteQuotationTags(xmlDocument, xmlDocumentCopy, extractQuoteHtml.quotationCheckpoints);
+
+    // Fix quirk in XmlDom.
+    if (xmlDocumentCopy.nodeType === NodeType.DOCUMENT_NODE && !xmlDocumentCopy.documentElement)
+      (xmlDocumentCopy.documentElement as any) = <HTMLElement>xmlDocumentCopy.childNodes[0];
+
+    // Serialize and return.
+    return {
+      body: xmlDomSerializer.serializeToString(xmlDocumentCopy, true),
+      didFindQuote: true
+    }
+  }
+  // Otherwise, if we found a known quote earlier, return the content before.
+  else if (!extractQuoteHtml.quoteWasFound && cutQuotations)
+    return { body: xmlDomSerializer.serializeToString(xmlDocumentCopy, true), didFindQuote: true };
+  // Finally, if no quote was found, return the original HTML.
+  else
+    return { body: messageBody, didFindQuote: false };
+}
+
+
+function extractQuoteHtmlViaMarkers(numberOfCheckpoints: number, xmlDocument: Document, ignoreBlockTags: Boolean): {
```

Comment:
Let's use an `options` object for the last argument rather than passing a single boolean.
(Otherwise, what it does is not obvious from the call site).